backends-velox/src/main/scala/org/apache/gluten/execution/python/ColumnarArrowEvalPythonExec.scala

```diff
@@ -341,11 +355,20 @@ case class ColumnarArrowEvalPythonExec(
               val joinedVectors = (0 until inputCb.numCols).toArray.map(
                 i => inputCb.column(i)) ++ (0 until outputCb.numCols).toArray.map(
                 i => outputCb.column(i))
+              // Columns in outputCb has random 0 or 1 refCnt and will fail checks in ensureOffload,
+              // so we do a hard reset here.
```

Comment:
Did you figure out why "random ref count" happens?

Reply:
Not quite. The outputCb is directly retrieved from python worker so can't debug why it happens.